bin/sozo/src/ops/auth.rs

```diff
@@ -16,25 +21,55 @@ pub async fn execute(command: AuthCommand, env_metadata: Option<Environment>) ->
 
                 let account = account.account(&provider, env_metadata.as_ref()).await?;
                 let world = WorldContract::new(world_address, &account);
+                let world_reader = WorldContractReader::new(world_address, &provider)
+                    .with_block(BlockId::Tag(BlockTag::Pending));
 
                 let mut calls = Vec::new();
 
                 for mc in models_contracts {
-                    let contract = get_contract_address(&world, mc.contract).await?;
-                    calls.push(world.grant_writer_getcall(&mc.model, &contract.into()));
+                    let model_name = parse_cairo_short_string(&mc.model)?;
+                    let model = world_reader.model_reader(&model_name).await?;
+                    let model_address = model.contract_address();
+
+                    if model_address == FieldElement::ZERO {
+                        println!(
+                            "{} Unknown model {} => {}",
+                            "[WARN]".yellow(),
+                            model_name.blue(),
+                            "IGNORED".yellow()
+                        );
+                    } else {
+                        let contract = get_contract_address(&world, mc.contract).await?;
+
+                        if model_address != contract {
+                            println!(
+                                "{} Wrong address for model {} => {}",
+                                "[WARN]".yellow(),
+                                model_name.blue(),
+                                "IGNORED".yellow()
+                            );
+                        } else {
```

Comment:
```suggestion
```
This check is not required here. With this, you are enforcing the contract (for which we want to grant access), **is** the model contract address. Which doesn't let room for any other contract to write data with this model.

We can't enforce any condition on the contract address (at least in this scenario), as it may be a system deployed on Dojo, but it can also be an account or any third party contract.

Reply:
Ah yes, this is a stupid check 😅
Thanks for the explanation, I remove it 👍